src/mono/mono/mini/aot-compiler.c

```diff
@@ -9138,6 +9138,9 @@ compile_method (MonoAotCompile *acfg, MonoMethod *method)
 		flags = (JitFlags)(flags | JIT_FLAG_SELF_INIT);
 	if (acfg->flags & MONO_AOT_FILE_FLAG_CODE_EXEC_ONLY)
 		flags = (JitFlags)(flags | JIT_FLAG_CODE_EXEC_ONLY);
+	if (acfg->dedup_collect_only)
+		/* Just collecting instances, no need to generate code */
+		flags = (JitFlags)(flags | JIT_FLAG_SKIP_CODEGEN);
```

Comment:
I don't understand why this only skips non-generic methods.  It looks like we set `dedup_collect_only` on all assemblies that are not the dedup_assembly

Reply:
The dedup step works by re-aot-ing everything, collecting instances, then throwing away the result. This just lets the JIT stop earlier, not generating code which will be thrown away anyway.